src/Queue/Client.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+declare(strict_types=1);
+
+namespace NS8\ProtectSDK\Queue;
+
+use NS8\ProtectSDK\Queue\Exceptions\Decoding as DecodingException;
+use NS8\ProtectSDK\Queue\Exceptions\Response as ResponseException;
+use Zend\Http\Client as ZendClient;
+use function array_key_exists;
+use function json_decode;
+use function json_last_error;
+use function sprintf;
+
+/**
+ * Manage updates from Queue for order info
+ */
+class Client extends BaseClient
+{
+    /**
+     * Attribute to track URL for connecting to queue
+     *
+     * @var string $url
+     */
+    protected static $url;
+
+    /**
+     * Attribute to track HTTTP Client used for sending requests
+     *
+     * @var ZendClient $httpClient
+     */
+    protected static $httpClient;
+
+    /**
+     * SQS Request Constants
+     */
+    public const DEFAULT_SQS_REQUEST_TYPE = 'GET';
+
+    /**
+     * Error keys used when receiving errors from SQS
+     */
+    public const ERROR_KEY         = 'Error';
+    public const ERROR_CODE_KEY    = 'Code';
+    public const ERROR_MESSAGE_KEY = 'Message';
+
+    /**
+     * Message keys used when parsing messages from SQS
+     */
+    public const RECEIVE_MESSAGE_RESPONSE_KEY = 'ReceiveMessageResponse';
+    public const RECEIVE_MESSAGE_RESULT_KEY   = 'ReceiveMessageResult';
+    public const RECEIVE_MESSAGE_SET_KEY      = 'messages';
+
+    /**
+     * Static headers that we are required to send in SQS requests
+     */
+    public const REQUIRED_HEADERS = [
+        'Accept'    => 'application/json',
+        'Host'      => 'sqs.us-west-2.amazonaws.com',
+    ];
+
+    /**
+     * Initializes the class to ensure attributes are set up correctly.
+     *
+     * @param ZendClient $httpClient The HTTP client used to make requests
+     * @param string     $queueUrl   URL used to access queue
+     *
+     * @return void
+     */
+    public static function initialize(?ZendClient $httpClient = null, ?string $queueUrl = null) : void
+    {
+        self::$url        = $queueUrl;
+        self::$httpClient = $httpClient ?? new ZendClient();
+    }
+
+    /**
+     * Returns the URL being used for queue iteration during runtime
+     *
+     * @return string The URL to be used for queue connection
+     */
+    public static function getQueueUrl() : string
+    {
+        if (! empty(self::$url)) {
+            return self::$url;
+        }
+
+        //TODO: This MUST be updated once queue URL lambda is in-place
+        self::$url = '';
```

Comment:
@crfroehlich @cwalv This will need to be updated and potentially have a method set-up once the lambda is in a position to be reached via an HTTP Request.